src/nrtk_explorer/app/transforms.py

```diff
@@ -112,6 +112,9 @@ def __init__(
         known_args, _ = self.server.cli.parse_known_args()
         self.state.inference_models = known_args.models
         self.state.object_detection_model = self.state.inference_models[0]
+        self.state.setdefault("image_list_ids", [])
+        self.state.setdefault("dataset_ids", [])
+        self.state.setdefault("user_selected_ids", [])
```

Comment:
I remember we talked about this last week and now I finally get it.  The reason the sub-apps broke is I don't dev/test with them, so me renamed/added shared state lead to the rot.   

A pattern we could migrate to that will make us more resilient: 
A "shared_setup.py" which core.py and the subapps depend on.  We use the `server.root_server == self.server` standalone check within shared_setup.setup(). 
What do you think?

Reply:
Good idea if we can do in each sub-app something like:

```python
nrtk_mixin.setup_dependency(
   "image_provider",
   "transformers", 
   "...",
)

```